File: Codeforces/Extra problems/1881d.cpp
#include <iostream>
#include <vector>
#include <cmath>

using namespace std;

int main() {
    long long int t;
    cin >> t;
    while (t--) {
        long long int n;
        cin >> n;
        vector<long long int> l(n);
        long long int prod = 1;
        for (long long int i = 0; i < n; i++) {
            cin >> l[i];
            prod *= l[i];
        }
        long long int num =(floor(pow(prod, 1.0 / n)));
        // cout<<num<<endl;
        if (pow(num, n) == prod || pow(num + 1, n) == prod) {
            cout << "YES" << endl;
        } else {
            cout << "NO" << endl;
        }
    }
    return 0;
}

File: Codeforces/Extra problems/1881d.py
import sys
input = sys.stdin.readline
from math import ceil,floor
for _ in range(int(input())):
    #n,m=map(int,input().split())
    n=int(input())
    #s=str(input())
    l=list(map(int,input().split()))
    prod=1
    for i in l:
        prod*=i
    # print(prod**(1/n),int(prod**(1/n)))
    num=floor(prod**(1/n))
    if(num**n==prod or (num+1)**n==prod):
        print("YES")
    else:
        print("NO")
    #l[:0]=s
